src/vaadin-grid-column.html

```diff
@@ -236,14 +236,17 @@
       footerCell.parentElement.hidden = false;
     }
 
-    _selectFirstTemplate(selector) {
-      return Array.prototype.filter.call(Polymer.dom(this).querySelectorAll(selector), el => {
-        return Polymer.dom(el).parentNode === this;
-      })[0];
+    _selectFirstTemplate(header = false, footer = false) {
+      return Polymer.FlattenedNodesObserver.getFlattenedNodes(this)
+        .filter(node => {
+          return node.localName === 'template'
+            && node.classList.contains('header') === header
+            && node.classList.contains('footer') === footer
+        })[0];
     }
 
-    _findTemplate(selector) {
-      const template = this._selectFirstTemplate(selector);
+    _findTemplate(header, footer) {
```

Comment:
Heads up @limonte. I believe a certain extension depends on this function to accept a String type parameter. This PR is targeting master but it's not guaranteed we wouldn't backport this to 4.x.

Reply:
I pushed the commit to `4.x` branch so we will be notified in case we port this change to `4.x`: https://github.com/vaadin/vaadin-grid/commit/babe9d837ed9bb599e9584ffcb9fb196ea98de7f